src/guest/vm_builder_qemu.cc

```diff
@@ -277,6 +281,22 @@ bool VmBuilderQemu::SetupSriov(void) {
     return true;
 }
 
+void VmBuilderQemu::SetPowerCtrl(void) {
```

Comment:
we already have void VmBuilderQemu::BuildGuestPmCtrlCmd(void), why need this?

Reply:
This socket needed for another functionality. This is used for suspend/hibernate power control feature.
Please refer https://jira.devtools.intel.com/browse/OAM-104253

Reply:
uhmm, so what's the difference? BuildGuestPmCtrlCmd() vs SetPowerCtrl(), they are really confusing me. Both functions are creating qmp socket, and are both for power managing, If they are just for different functions, can they be merged?

Reply:
According to design, BuildGuestPmCtrlCmd() needs to execute only if it is enabled from the config ini file. SetPowerCtrl() needs to run irrespective of the mentioned config in ini file.

Reply:
The new requirement is needed to suspend from the host side. The host can suspend the VMs. The two features cannot share the same qmp socket. The new socket is a place-holder for now as it is not yet inetegrated in CIV.

Reply:
> According to design, BuildGuestPmCtrlCmd() needs to execute only if it is enabled from the config ini file. SetPowerCtrl() needs to run irrespective of the mentioned config in ini file.

why it should be run irrespectively. I don't think it should be run by default, it should be an addon.

Reply:
> 

if it not implemented yet, please pending merge, we need to clarify the detail and make the code clean. you can seperate this PR to 2 parts, we can merge the first part.